lib/main.coffee

```diff
@@ -49,14 +38,18 @@ module.exports =
         filePath = textEditor.getPath()
         command = @executablePath
         ruleset = @rulesets
-        if @projectRules
-          rulesetPath = helpers.find(filePath, 'ruleset.xml')
+        if /^[a-z0-9]+\.xml$/gi.test(@rulesets)
+          rulesetPath = helpers.find(filePath, @rulesets)
           ruleset = rulesetPath if rulesetPath?
         parameters = []
         parameters.push(filePath)
         parameters.push('text')
         parameters.push(ruleset)
-        return helpers.exec(command, parameters).then (output) ->
+        options = {}
+        projectDir = atom.project.relativizePath(filePath)[0]
+        if projectDir?
+          options.cwd = projectDir
```

Comment:
If you had a `ruleset.xml` in the same directory as the file, but had for some bizarre reason opened the file in Atom outside a project context it would then still find it if you had something along the lines of `codesize,ruleset.xml`, where otherwise it wouldn't since it would be running with a cwd of the Atom directory.

(I think, haven't actually tested)